### ISSUE TITLE
How to work with @isDate annotation

### ISSUE THREAD

**Issue by stephannielsen** [lukeautry/tsoa#1350]
## Sorting

- **I'm submitting a ...**

  - [ ] bug report
  - [ ] feature request
  - [x] support request

- I confirm that I
  - [x] used the [search](https://github.com/lukeautry/tsoa/search?type=Issues) to make sure that a similar issue hasn't already been submit

## Expected Behavior

When using the `@isDate` annotation with a `Date` type, the generated OpenAPI spec looks fine, the prop is defined as `string($date)` and the examples work correctly.

```ts
export interface Response {
    /**
     * @isDate
     */
    someDate: Date;
}
```

But when we use `Date` on TypeScript side, the value returned from the API is a full date ISO string, `2022-11-24T00:00:00Z`. This can be fixed by formatting the `Date` value manually to just a date string, `2022-11-24` and returning this, but that also means the interface needs to change as a `string`:

```ts
export interface Response {
    /**
     * @isDate
     */
    someDate: string;
}
```

Now the API behaves correctly by just returning the date, but the OpenAPI spec is not correct anymore. The `@isDate` annotation has no effect, the prop is just defined as `string` and the examples don't work anymore either.

I understand that TSOA is not performing formatting, so I don't think it's a bug in TSOA. But I am not sure how to leverage the `@isDate` annotation now correctly for **response models**. Is there another way how to generate a correct response model in the OpenAPI spec for dates?

## Current Behavior

Currently, it seems not possible to declare a property to be of date format correctly for OpenAPI spec for responses. See above.

## Context (Environment)

Version of the library: 4.1.3
Version of NodeJS: v18 LTS

- Confirm you were using yarn not npm: [x]

## Detailed Description

Allow `@isDate` annotation to be used on `string` props and correctly infer OpenAPI spec from it with examples and so on.

## Breaking change?

No, I don't think it's a breaking change.

**Comment by stephannielsen** [comment]
Ok, I guess I found the solution 😄 

```ts
/**
   * @format date
   **/
  someDate: string
```
With this, the spec shows the type correctly and examples work too. Would be nice to work `@isDate` in the same way though to keep it consistent?

**Comment by WoH** [comment]
https://tsoa-community.github.io/docs/annotations.html#list-of-supported-keywords-with-arguments

Have you checked out the docs here?

**Comment by stephannielsen** [comment]
Hm, yes I did. But with just `@isDate`/`@isDateTime` a `Date` prop also shows up in the OpenAPI spec correctly as `$string($date)` or `$string($date-time)` without the `@format` annotation just fine. But only if the property type is `Date`, not if it is `string`.

I figured it out when digging deeper into the linked OpenAPI spec documentation: https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.3.md#properties and the explanation of `format` and the possible data types.
